src/builtin_string.cpp

```diff
@@ -1244,6 +1255,83 @@ static int string_collect(parser_t &parser, io_streams_t &streams, int argc, wch
     return buff.size() > 0 ? STATUS_CMD_OK : STATUS_CMD_ERROR;
 }
 
+static int string_pad(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv) {
+    wchar_t *cmd = argv[0];
+
+    options_t opts;
+    opts.char_to_pad_valid = true;
+    opts.count_valid = true;
+    opts.max_valid = true;
+    opts.left_valid = true;
+    opts.right_valid = true;
+    opts.quiet_valid = true;
+    int optind;
+    int retval = parse_opts(&opts, &optind, 0, argc, argv, parser, streams);
+    if (retval != STATUS_CMD_OK) return retval;
+
+    // If neither left or right is specified, we pad only on the left.
+    if (!opts.left && !opts.right) {
+        opts.left = true;
+        opts.right = false;
+    }
+
+    // If both count and max is specified, we raise an error.
+    if (opts.count && opts.max) {
+        streams.err.append_format(BUILTIN_ERR_COMBO2, cmd,
+                                  _(L"--count and --max are mutually exclusive"));
+        return STATUS_INVALID_ARGS;
+    }
+
+    size_t npad = 0;
+
+    arg_iterator_t aiter(argv, optind, streams);
+    while (const wcstring *arg = aiter.nextstr()) {
+        wcstring padded = wcstring(*arg);
+        size_t pad_left = 0;
+        size_t pad_right = 0;
+
+        if (opts.count && opts.right) {
+            pad_right = opts.count;
+        }
+
+        if (opts.count && opts.left) {
+            pad_left = opts.count;
+        }
+
+        if (opts.max >= padded.size()) {
+            if (opts.left && opts.right) {
+                // pad_left will get one more character
+                pad_left = (opts.max - padded.size() + 1) / 2;
+                pad_right = (opts.max - padded.size()) / 2;
+            }
+
+            if (!opts.left && opts.right) {
+                pad_right = opts.max - padded.size();
+            }
+
+            if (opts.left && !opts.right) {
+                pad_left = opts.max - padded.size();
+            }
+        }
+
+        if (pad_left) {
+            padded.insert(0, pad_left, opts.char_to_pad);
+        }
+
+        if (pad_right) {
```

Comment:
The `if` seems redundant - I believe nothing happens if `pad_right` is zero. Same above.